### ISSUE TITLE
Should `uploadStatus` be kept in model?

### ISSUE THREAD

**Issue by scofalik** [ckeditor/ckeditor5-upload#40]
At the moment, upload status for a file is kept in as `uploadStatus` attribute on image element.

From the first sight this solution felt bad to me. Is upload status really a piece of data? Is state of element something that should be kept in model? I am fine with `uploadId` because this characterises an image, just like `src` do. But `uploadStatus`?

I understand that we want to represent upload status changes in view and using model and conversion was the easiest and tested way to do it, but I feel like we shouldn't have this in model.

It already created a problem with undo. As I wrote, upload status does not feel like data and undoing it is completely wrong as upload status should be set and controlled only by the part of code that is responsible for uploading. I overcome this by setting `uploadStatus` in transparent batches.

So I open this for discussion.

**Comment by szymonkups** [comment]
I think it would be possible to get rid of `uploadStatus` attribute if it causes problems. It's a good point that it feels that its place is not in the model. 

Currently, `ImageUploadEngine` checks if image with `uploadId` is inserted and starts handling the upload process in the model. It sets correct `uploadStatus` and `src` when upload is finished. Then, plugins like `ImageUploadProgress` registers model to view converters which listens to `uploadStatus` changes and update view accordingly.

I think it can be done without `uploadStatus` attribute. We need to remove any `uploadStatus` handling from `ImageUploadEngine`. `ImageUploadProgress` could now also listen if any image with `uploadId` is inserted (probably with lower priority than `ImageUploadEngine`) and base it's changes in the view on status taken directly from `FileLoader` associated with image's `uploadId`.

**Comment by scofalik** [comment]
Yes I think that would you described would solve this "issue", however, as I said, I already fixed it by moving those `uploadStatus` changes to transparent batches. So we don't have any bugs now, I left this issue "for future" :)

**Comment by pjasiun** [comment]
I agree that binding between `uploadId` and `uploadStatus` could be done outside the model, in some controller. AFAIR, it was done in the model, because we wanted to make passing it from the command to the plugin simple, but if it causes a problem we can pass it some other way or create a dedicated class to handle upload statuses.
